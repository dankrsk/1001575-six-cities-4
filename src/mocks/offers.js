const TYPES = [
  `Apartment`,
  `Room`,
  `House`,
  `Hotel`,
];

const TITLES = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
  `Wood and stone place`,
];

const getOffer = () => {
  return {
    id: Math.round(Math.random() * 10000),
    type: TYPES[Math.round(Math.random() * 3)],
    title: TITLES[Math.round(Math.random() * 4)],
    price: Math.round(Math.random() * 100),
    rating: Math.round(Math.random() * 5),
    isPremium: Math.random() > 0.5 ? true : false,
    photo: `http://cdn.home-designing.com/wp-content/uploads/2013/11/1-Contemporary-architecture.jpg`,
  };
};

export const getOffers = (count) => {
  return Array(count).fill(``).map(() => getOffer());
};
