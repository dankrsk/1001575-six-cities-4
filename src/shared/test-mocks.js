const imgUrl = `http://cdn.home-designing.com/wp-content/uploads/2013/11/1-Contemporary-architecture.jpg`;

export const mock = {
  offers: [
    {
      bedrooms: 3,
      city: `Amsterdam`,
      cityLocation: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      goods: [`Heating`, `Kitchen`],
      host: {
        avatarUrl: imgUrl,
        id: 1,
        isPro: true,
        name: `Dan`,
      },
      id: 1,
      images: [imgUrl, imgUrl],
      isFavorite: true,
      isPremium: true,
      coordinates: [52.35514938496378, 4.673877537499948],
      maxAdults: 4,
      photo: imgUrl,
      price: 100,
      rating: 1,
      title: `Beautiful & luxurious apartment at great location`,
      type: `House`,
    },
    {
      bedrooms: 3,
      city: `Cologne`,
      cityLocation: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      goods: [`Heating`, `Kitchen`],
      host: {
        avatarUrl: imgUrl,
        id: 2,
        isPro: true,
        name: `Dan`,
      },
      id: 2,
      images: [imgUrl, imgUrl],
      isFavorite: true,
      isPremium: true,
      coordinates: [52.35514938496378, 4.673877537499948],
      maxAdults: 4,
      photo: imgUrl,
      price: 100,
      rating: 1,
      title: `Beautiful & luxurious apartment at great location`,
      type: `Apartment`,
    },
    {
      bedrooms: 3,
      city: `Berlin`,
      cityLocation: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      goods: [`Heating`, `Kitchen`],
      host: {
        avatarUrl: imgUrl,
        id: 3,
        isPro: true,
        name: `Dan`,
      },
      id: 3,
      images: [imgUrl, imgUrl],
      isFavorite: true,
      isPremium: true,
      coordinates: [52.35514938496378, 4.673877537499948],
      maxAdults: 4,
      photo: imgUrl,
      price: 100,
      rating: 1,
      title: `Beautiful & luxurious apartment at great location`,
      type: `Room`,
    },
  ]
};
