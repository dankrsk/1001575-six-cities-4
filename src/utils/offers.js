const CITIES_COUNT = 6;

export const getCitiesFromOffers = (offers) => {
  const cities = [];
  let i = 0;

  while (cities.length < CITIES_COUNT && i < offers.length) {
    const city = offers[i].city;

    if (!cities.includes(city)) {
      cities.push(city);
    }

    i++;
  }

  return cities;
};
