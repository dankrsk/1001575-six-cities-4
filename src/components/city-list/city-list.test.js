import React from 'react';
import renderer from 'react-test-renderer';
import CityList from '../city-list/city-list.jsx';
import {mock} from '../../shared/test-mocks.js';
import {getCitiesFromOffers} from '../../utils/offers.js';

describe(`Snapshots for CityList`, () => {
  it(`CityList component`, () => {
    const tree = renderer
      .create(<CityList
        currentCity={`Paris`}
        allCities={getCitiesFromOffers(mock.offers)}
        onCityLinkClick={() => {}}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
