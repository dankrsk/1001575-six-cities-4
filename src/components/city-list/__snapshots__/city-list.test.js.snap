// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots for CityList CityList component 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>,
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Berlin
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>,
]
`;
